server/webapp/WEB-INF/rails.new/app/controllers/api_v1/admin/pluggable_scms_controller.rb

```diff
@@ -0,0 +1,86 @@
+##########################################################################
+# Copyright 2016 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV1
+  module Admin
+    class PluggableScmsController < ApiV1::BaseController
+      before_action :check_admin_user_and_401
+      before_action :check_for_stale_request, :check_for_scm_rename, only: [:update]
+
+
+      def index
+        scms = pluggable_scm_service.listAllScms
+        render json_hal_v1: ApiV1::Scms::PluggableScmsRepresenter.new(scms).to_hash(url_builder: self)
+      end
+
+      def show
+        hash = get_scm_hash(params[:material_name])
+        render json_hal_v1: hash if stale?(etag: JSON.generate(hash))
```

Comment:
Please use `render DEFAULT_FORMAT => ...` instead of `render json_hal_v1: ...` — makes it easy to upgrade to api v2 if needed :)